internal/rbac/interactor.go

```diff
@@ -0,0 +1,35 @@
+package rbac
+
+import (
+	"context"
+
+	"github.com/SatorNetwork/sator-api/internal/jwt"
+)
+
+func CheckRole(userRole Role, requiredRoles ...Role) error {
+	for _, v := range requiredRoles {
+		switch v {
+		case userRole, AvailableForAllRoles:
+			return nil
+		case AvailableForAuthorizedUsers:
+			if userRole != RoleGuest {
+				return nil
+			}
+		}
+	}
+
+	return ErrAccessDenied
+}
+
+func CheckRoleFromContext(ctx context.Context, requiredRoles ...Role) error {
```

Comment:
it seems that `CheckRoleFromContext` never returns error? Because `requiredRoles ` param is ignored and `CheckRole(Role(role), AvailableForAllRoles)` always returns nil